conftest.py

```diff
@@ -31,6 +31,16 @@ def pytest_sessionstart():
     downloader.main(['-d', 'sct_testing_data', '-o', sct_test_path()])
 
 
+@pytest.fixture
+def run_in_sct_testing_data_dir():
+    """Temporarily change the working directory to 'sct_testing_data'. This replicates the behavior of the old
+    `sct_testing`, and is needed to prevent tests from cluttering the working directory with output files."""
+    cwd = os.getcwd()
+    os.chdir(sct_test_path())
+    yield
```

Comment:
If you do this

```suggestion
def test_data():
    """Get the path to the test data folder.
    
    Temporarily changes the working directory to 'sct_testing_data'. This replicates the behavior of the old
    `sct_testing`, and is needed to prevent tests from cluttering the working directory with output files
    but should not be relied on in new tests."""
    cwd = os.getcwd()
    os.chdir(sct_test_path())
    yield pathlib.Path(os.getcwd())
```

then you can start improving the tests with e.g.

```
im_gt = Image(test_data/'mt/mt1.nii.gz')
```

and then it will be easier to get rid of the `chdir` eventually.

Reply:
And it also shortens the fixture name. I think fixtures are supposed to be nouns, not verbs, generally.

Reply:
Same feelings here about wanting to handle this as part of #3422. I definitely agree that there are improvements to be made re: `sct_testing_data`, but I figure we can standardize the test suite as a whole in a dedicated PR.